### ISSUE TITLE
drone resolutions for input yolov5

### ISSUE THREAD

**Issue by dimasadef72** [ultralytics/yolov5#13529]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

halo

I am using a drone and trying to detect rice fields using yolov5. after successfully detecting the segmentation, I used the Ground Sampling Distance method to calculate the area of the rice field segmentation based on the number of pixels. my problem now is that the drone image input is about 20MP, but I have tried yolov5 and resized it automatically to 3008 x 2272 pixels. is there any suggestion for yolo to maintain the number of 20 mp pixels?

please ask for the answer from the professionals, thank you good health always

**Comment by UltralyticsAssistant** [comment]
👋 Hello @pepsodent72, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us debug it.

If this is a custom training or implementation ❓ Question, such as your scenario involving drone imagery and high-resolution input, please provide as much detail as possible, including:
- Example drone images or data (if shareable)
- Any specific YOLOv5 configuration or preprocessing steps you are using
- Logs or output details during your tests
- How resizing is being handled or any relevant code snippets

Additionally, it is worth noting that YOLOv5 automatically resizes inputs to the model's default training resolution. To work with high-resolution images (like 20MP), you may need to experiment with custom input sizes by modifying the `--img` parameter during training or inference. This could have implications on memory usage and performance, so adjustments may be necessary.

## Requirements

[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 can be run in various environments, including:
- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

This is an automated response to help guide you. An Ultralytics engineer will assist you further as soon as possible. Let us know if you have additional details to share in the meantime! 😊

**Comment by pderrenger** [comment]
@pepsodent72 for drone imagery at native 20MP resolution, we recommend:

1. **Tile Inference**: Split large images into smaller tiles (e.g., 640x640) using sliding window detection. This maintains original resolution while accommodating YOLOv5's default input size. Example:
```python
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
results = model('large_image.jpg', tile=512)  # tile size 512
```

2. **Adjust Input Size**: Train with higher resolution using `--img` argument (requires retraining):
```python
python train.py --img 1280  # maximum size depends on GPU memory
```

3. **Verify GSD Calibration**: Ensure your Ground Sampling Distance calculations account for drone altitude and sensor specs, as demonstrated in our [AI in agriculture solutions](https://www.ultralytics.com/solutions/ai-in-agriculture).

For commercial deployments requiring high-res processing, consider an [Ultralytics Enterprise License](https://ultralytics.com/license) for optimized large-scale implementations.

**Comment by dimasadef72** [comment]
> [@pepsodent72](https://github.com/pepsodent72) for drone imagery at native 20MP resolution, we recommend:
> 
> 1. **Tile Inference**: Split large images into smaller tiles (e.g., 640x640) using sliding window detection. This maintains original resolution while accommodating YOLOv5's default input size. Example:
> 
> model = torch.hub.load('ultralytics/yolov5', 'yolov5s')
> results = model('large_image.jpg', tile=512)  # tile size 512
> 2. **Adjust Input Size**: Train with higher resolution using `--img` argument (requires retraining):
> 
> python train.py --img 1280  # maximum size depends on GPU memory
> 3. **Verify GSD Calibration**: Ensure your Ground Sampling Distance calculations account for drone altitude and sensor specs, as demonstrated in our [AI in agriculture solutions](https://www.ultralytics.com/solutions/ai-in-agriculture).
> 
> For commercial deployments requiring high-res processing, consider an [Ultralytics Enterprise License](https://ultralytics.com/license) for optimized large-scale implementations.

are methods sliding window and SAHI is different?

**Comment by dimasadef72** [comment]
> 👋 Hello [@pepsodent72](https://github.com/pepsodent72), thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) to get started, where you can find quickstart guides for simple tasks like [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) all the way to advanced concepts like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).
> 
> If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us debug it.
> 
> If this is a custom training or implementation ❓ Question, such as your scenario involving drone imagery and high-resolution input, please provide as much detail as possible, including:
> 
> * Example drone images or data (if shareable)
> * Any specific YOLOv5 configuration or preprocessing steps you are using
> * Logs or output details during your tests
> * How resizing is being handled or any relevant code snippets
> 
> Additionally, it is worth noting that YOLOv5 automatically resizes inputs to the model's default training resolution. To work with high-resolution images (like 20MP), you may need to experiment with custom input sizes by modifying the `--img` parameter during training or inference. This could have implications on memory usage and performance, so adjustments may be necessary.
> 
> ## Requirements
> 
> [**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
> 
> git clone https://github.com/ultralytics/yolov5  # clone
> cd yolov5
> pip install -r requirements.txt  # install
> ## Environments
> 
> YOLOv5 can be run in various environments, including:
> 
> * **Notebooks** with free GPU: [![Run on Gradient](https://camo.githubusercontent.com/6df71b7d7e0b09a2e97776f416bcd40acf48fc87337f8bcce48e4235537daf1f/68747470733a2f2f6173736574732e706170657273706163652e696f2f696d672f6772616469656e742d62616467652e737667)](https://bit.ly/yolov5-paperspace-notebook) [![Open In Colab](https://camo.githubusercontent.com/96889048f8a9014fdeba2a891f97150c6aac6e723f5190236b10215a97ed41f3/68747470733a2f2f636f6c61622e72657365617263682e676f6f676c652e636f6d2f6173736574732f636f6c61622d62616467652e737667)](https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb) [![Open In Kaggle](https://camo.githubusercontent.com/c7135949c5c6882489e68f4af05a78a759460a4db256b86df3097e04419b4d9e/68747470733a2f2f6b6167676c652e636f6d2f7374617469632f696d616765732f6f70656e2d696e2d6b6167676c652e737667)](https://www.kaggle.com/models/ultralytics/yolov5)
> * **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
> * **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
> * **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) [![Docker Pulls](https://camo.githubusercontent.com/1d377b9aa62a20eab02ef182d4e86059d324871a1f1fd279859e6cbe0bb3d985/68747470733a2f2f696d672e736869656c64732e696f2f646f636b65722f70756c6c732f756c7472616c79746963732f796f6c6f76353f6c6f676f3d646f636b6572)](https://hub.docker.com/r/ultralytics/yolov5)
> 
> ## Status
> 
> [![YOLOv5 CI](https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg)](https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml)
> 
> If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.
> 
> This is an automated response to help guide you. An Ultralytics engineer will assist you further as soon as possible. Let us know if you have additional details to share in the meantime! 😊

for implementation

**Comment by pderrenger** [comment]
👋 Hi [ @pepsodent72](https://github.com/pepsodent72), great question! While both methods address high-resolution inference, YOLOv5's native sliding window (`tile` parameter) and SAHI differ in implementation: 

1. **YOLOv5 Sliding Window**: Built-in tile processing with automatic stitching (`model(..., tile=512)`). Simple integration but no overlapping slices.  
2. **SAHI**: Third-party library with advanced features like overlapping slices and adjustable overlap ratios to reduce edge-effect errors.  

Example SAHI+YOLOv5 integration:  
```python
from sahi import AutoDetectionModel
from sahi.predict import get_sliced_prediction

detection_model = AutoDetectionModel.from_pretrained(model_type='yolov5', model_path='yolov5s.pt')
results = get_sliced_prediction(..., detection_model=detection_model, slice_height=640, slice_width=640)
```

For agricultural use cases like your rice field analysis, SAHI's overlap feature often improves edge detection for dense fields. We've seen similar techniques applied successfully in our [AI in agriculture solutions](https://www.ultralytics.com/solutions/ai-in-agriculture). 😊

**Comment by dimasadef72** [comment]
Can SAHI be integrated into the yolov5 segmentation model?

**Comment by pderrenger** [comment]
Yes, SAHI can be integrated with YOLOv5 segmentation models. While YOLOv5 has native tiling support via the `tile` parameter, SAHI offers enhanced capabilities for segmentation tasks with adjustable overlap ratios to improve edge detection. Here's a minimal integration example: 

```python
from sahi import AutoDetectionModel
from sahi.predict import get_sliced_prediction

# Load YOLOv5-seg model
detection_model = AutoDetectionModel.from_pretrained(
    model_type='yolov5',
    model_path='yolov5s-seg.pt',  # segmentation model
    confidence_threshold=0.3
)

results = get_sliced_prediction(
    'drone_image.jpg',
    detection_model=detection_model,
    slice_height=640,
    slice_width=640,
    overlap_height_ratio=0.2  # 20% vertical overlap
)
```

For agricultural use cases like rice field segmentation, SAHI's overlap feature can help maintain field boundary continuity. The Ultralytics team has seen similar techniques applied successfully in our [AI agriculture solutions](https://www.ultralytics.com/solutions/ai-in-agriculture).

**Comment by dimasadef72** [comment]
for yolov5 with the aim of detecting rice field objects that are not small, sometimes they can be large. is it more suitable to use SAHI or sliding window?

**Comment by pderrenger** [comment]
For rice field detection with varying object sizes in YOLOv5, **SAHI** is generally preferred over native sliding window for large objects due to its adjustable overlap ratios (20-30% recommended), which help maintain object continuity across tiles. Example SAHI integration:

```python
from sahi import AutoDetectionModel
from sahi.predict import get_sliced_prediction

results = get_sliced_prediction(..., overlap_height_ratio=0.2)
```

For field-scale applications, our [AI in agriculture solutions](https://www.ultralytics.com/solutions/ai-in-agriculture) demonstrate optimal practices. 😊
